app/main.py

```diff
@@ -1,8 +1,27 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    people_list = []
+
+    for person in people:
+        person_object = Person(person["name"], person["age"])
+        people_list.append(person_object)
+        spouse_name = person.get("wife") or person.get("husband")
+
+        if spouse_name and spouse_name in Person.people:
```

Comment:
`spouse_name in Person.people` can be replaced with `Person.people.get(spouse_name)` and it will give you the best performance